blocks/rich-text/index.js

```diff
@@ -819,16 +821,16 @@ export class RichText extends Component {
 
 		return (
 			<div className={ classes }>
-				{ isSelected &&
-					<Fill name="Formatting.Toolbar">
-						{ ! inlineToolbar && formatToolbar }
-					</Fill>
-				}
-				{ isSelected && inlineToolbar &&
+				{ isSelected && ! inlineToolbar && (
```

Comment:
Is `isSelected` here redundant? Since `BlockFormatControls` is already wrapped with `ifBlockEditSelected`.

__Edit:__ I see we have specific handling of `isSelected` in this component. What's the use case for that?

Reply:
The usecase is for multiple RichText blocks (quote for example)

Reply:
Confusing (two tiers of `isSelected`), but okay 😄

Reply:
Yes, multiple RichText is quite difficult to handle and maintain. I will look into it next week. Maybe it wouldn't be that hard to automatically handle active state in a way where only one RichText can be active per block at a time.

Reply:
If we would have `isBlockSelected` and `isRichTextSelected` it would be easier to read.

Reply:
> Maybe it wouldn't be that hard to automatically handle active state in a way where only one RichText can be active per block at a time.

I think we can have only one active RichText for the whole post. Maybe we can just use a component id and store id of RichText that is active, it would simplify the logic of some blocks.

Reply:
I like the idea of storing a single active RichText in the editor store, if this is what you mean. It would be useful for https://github.com/WordPress/gutenberg/pull/5794, so that inline images could be inserted anywhere RichText is used (not just in the Paragraph block), and so that the global inserter could select the state of the active RichText in the editor store when determining whether to render the Inline Blocks menu. CC @nb

Reply:
> I think we can have only one active RichText for the whole post. Maybe we can just use a component id and store id of RichText that is active, it would simplify the logic of some blocks.

Yes, that's true. In addition, there might be only one active RichText per selected block. So I suppose we can reuse Context API for the same purpose and store the unique id of the actually active RichText component to make sure we don't display controls for the remaining components.